torchao/quantization/quant_api.py

```diff
@@ -1807,17 +1837,20 @@ def _float8_dynamic_activation_float8_weight_quantize_tensor(weight, config):
     activation_value_lb = config.activation_value_lb
     activation_value_ub = config.activation_value_ub
     kernel_preference = config.kernel_preference
+    float8_packing_format = config.float8_packing_format
 
     # Ensure works on device
-    _check_hardware_support(granularity)
+    is_cpu = weight.device.type == "cpu"
+    if not is_cpu:
+        _check_hardware_support(granularity)
```

Comment:
can you move this to version 1? and then version 2 can do this check in the tensor itself probably